nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -409,24 +590,44 @@ in {
       (mkIf cfg.netbootxyz.enable {
         "efi/netbootxyz/netboot.xyz.efi" = "${pkgs.netbootxyz-efi}";
       })
+      (mkIf (cfg.edk2-uefi-shell.enable || cfg.windows != { }) {
+        ${edk2ShellEspPath} = "${pkgs.edk2-uefi-shell}/shell.efi";
+      })
     ];
 
-    boot.loader.systemd-boot.extraEntries = mkMerge [
-      (mkIf cfg.memtest86.enable {
-        "memtest86.conf" = ''
-          title  Memtest86+
-          efi    /efi/memtest86/memtest.efi
-          sort-key ${cfg.memtest86.sortKey}
+    boot.loader.systemd-boot.extraEntries = mkMerge (
+      [
+        (mkIf cfg.memtest86.enable {
+          "memtest86.conf" = ''
+            title  Memtest86+
+            efi    /efi/memtest86/memtest.efi
+            sort-key ${cfg.memtest86.sortKey}
+          '';
+        })
+        (mkIf cfg.netbootxyz.enable {
+          "netbootxyz.conf" = ''
+            title  netboot.xyz
+            efi    /efi/netbootxyz/netboot.xyz.efi
+            sort-key ${cfg.netbootxyz.sortKey}
+          '';
+        })
+        (mkIf cfg.edk2-uefi-shell.enable {
+          "edk2-uefi-shell.conf" = ''
+            title  EDK2 UEFI Shell
+            efi    /${edk2ShellEspPath}
+            sort-key ${cfg.edk2-uefi-shell.sortKey}
+          '';
+        })
+      ]
+      ++ (mapAttrsToList (winVersion: cfg: {
+        "windows_${winVersion}.conf" = ''
+          title ${cfg.title}
+          efi /${edk2ShellEspPath}
+          options -nointerrupt -nomap -noversion ${cfg.efiDeviceHandle}:EFI\Microsoft\Boot\Bootmgfw.efi
```

Comment:
Shouldn't this have been `bootmgfw.efi`, lowercase? I know the ESP is *usually* case insensitive, but there have definitely been cases where that wasn't true, whether that's because of buggy firmware or custom UEFI FS drivers.

Reply:
Well, on my PC at least the path is created by Windows exactly like this, in capital case.

Reply:
Huh, interesting. On mine it's lower case. I wonder why

Reply:
Might it be mounted case-insensitively on your machine and case-sensitively on mine?

Reply:
It definitely is mounted case-insensitively on my machine, but I'm pretty sure that doesn't affect the case of the file names you see when listing directories.

Reply:
If you add `boot.loader.systemd-boot.edk2-uefi-shell.enable = true;` to your config and boot into the edk2 shell, and then run `map -c` first and then for each of the device handles run  `ls HD0c1:\EFI`, do the paths show up as case-sensitive then?

Reply:
Ah I didn't know that was the case.